dbms/src/Storages/IStorage.h

```diff
@@ -82,6 +82,8 @@ class IStorage : public std::enable_shared_from_this<IStorage>
     /// Returns true if the storage supports deduplication of inserted data blocks.
     virtual bool supportsDeduplication() const { return false; }
 
+    using ColumnSizeByName = std::unordered_map<std::string, ColumnSize>;
+    virtual ColumnSizeByName getColumnSizes() const { return {}; }
```

Comment:
A comment that it's optional and the result is intended to use for query optimization.